src/jit/lowerxarch.cpp

```diff
@@ -925,7 +925,8 @@ void Lowering::TreeNodeInfoInit(GenTree* tree)
     // the below logic leads to conservative register specification.  In future when or if we find
     // one such case, this logic needs to be fine tuned for that case(s).
     if (varTypeIsByte(tree) || ((tree->OperGet() == GT_CAST) && varTypeIsByte(tree->CastToType())) ||
-        (tree->OperIsCompare() && varTypeIsByte(tree->gtGetOp1()) && varTypeIsByte(tree->gtGetOp2())))
+        (tree->OperIsCompare() && varTypeIsByte(tree->gtGetOp1()) &&
+         (varTypeIsByte(tree->gtGetOp2()) || tree->gtGetOp2()->IsCnsIntOrI())))
```

Comment:
> varTypeIsByte covers the following types TYP_BOOL, BYTE, UBYTE whose original size is 1-byte

Excellent.

> CHAR, SHORT, USHORT are 16-bit (or 2-byte) types - these don't need to be covered by these conditions.

Ah, yeah--I could not recall whether or not all x86 registers were 16-bit-addressable, but I believe they are. LGTM.